File: TelerikUIExamplesInSwift/TelerikUIExamplesInSwift/PasswordValidator.swift
//
//  PasswordValidator.swift
//  TelerikUIExamplesInSwift
//
//  Copyright (c) 2015 Telerik. All rights reserved.
//

class PasswordValidator: NSObject, TKDataFormValidator {

    var shortPassword: Bool?

    func validateProperty(property: TKEntityProperty) -> Bool {
        shortPassword = false
        let password: NSString = property.valueCandidate as! NSString
        if(password.length < 6) {
            shortPassword = true
            return false
        }
        
        return true
    }
    
    var validationMessage: String? {
        if(shortPassword == true) {
            return "Password must be at least 6 characters"
        }
        return nil
    }
}

File: TelerikUIExamplesInCSharp/Examples/DataForm/Classes/PasswordValidator.cs
using System;
using TelerikUI;
using Foundation;

namespace Examples
{
	public class PasswordValidator : TKDataFormValidator
	{
		bool shortPassword;

		public PasswordValidator ()
		{
		}

		public override string ValidationMessage {
			get {
				if (shortPassword) {
					return "Password must be at least 6 characters";
				}

				return null;
			}
		}

		public override bool ValidateProperty (TKEntityProperty property)
		{
			shortPassword = false;
			NSString password = (NSString)property.ValueCandidate;
			if (password.Length < 6) {
				shortPassword = true;
				return false;
			}

			return true;
		}

	}
}

